src/ng/compile.js

```diff
@@ -3101,6 +3152,33 @@ function $CompileProvider($provide, $$sanitizeUriProvider) {
         }
       });
 
+      function recordChanges(key, currentValue, previousValue) {
+        if (isFunction(destination.$onChanges) && currentValue !== previousValue) {
+          // If we have not already scheduled the top level onChangesQueue handler then do so now
+          if (!onChangesQueue) {
+            scope.$$postDigest(flushOnChangesQueue);
```

Comment:
Why `$$postDigest` versus `$evalAsync`?  Didn't pull this down to test, but `$evalAsync` would feel like the more correct method to use here so that we continue to live within the same digest cycle and don't accidentally violate the max TTL due to handling record changes